types/index.d.ts

```diff
@@ -343,7 +343,13 @@ declare namespace Realm {
         code: number;
     }
 
-    type ErrorCallback = (session: App.Sync.Session, error: SyncError) => void;
+    interface ClientResetError {
+        name: string;
+        path: string;
+        config: SyncConfiguration;
+    }
```

Comment:
I think this would be more correct? And should make it easier to determine if it's a a `ClientResetError` or a generic `SyncError`.
```suggestion
    interface ClientResetError {
        name: "ClientReset";
        path: string;
        config: SyncConfiguration;
        readOnly: true;
    }
```